src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/SuccessfulRetryTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ServiceControlCompatibilityTests
+{
+    using Autofac;
+    using NServiceBus;
+    using NUnit.Framework;
+    using System;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class SuccessfulRetryTests : SqlScTest
+    {
+        [TestCaseSource(nameof(AllTransports))]
+        public async Task Can_successfully_retry_a_failed_message(Type transportDetailType)
+        {
+            var transportDetails = StartUp(transportDetailType);
+            await RunTest(transportDetails);
+        }
+
+        async Task RunTest(ITransportDetails transportDetails)
+        {
+            var handlerInstance = new TestMessageHandler();
+            var builder = new ContainerBuilder();
+            builder.RegisterInstance(handlerInstance);
+            var container = builder.Build();
```

Comment:
I'm not sure. Let me see what I can do. This is the Arrange part of the AAA, and honestly I don't like the singleton nature here anyway, but couldn't come up with a nicer way to manage state within the handler reliably. I'll try work on it some more today.